### PR TITLE
Enable StrictMode for debug builds.

### PR DESCRIPTION
**Pull request by twaddington**
This logs policy violations such as performing network or disk I/O on the main thread.

### CODE REVIEW

**Comment by twaddington** [comment]
I noticed we're performing some disk and network operations on the main thread. For now, this just adds extra logging (see example below). However, we can make it more annoying, if we want to, by popping up a dialog or even crashing the application.

```
D/StrictMode: StrictMode policy violation; ~duration=1370 ms: android.os.strictmode.DiskReadViolation
        at android.os.StrictMode$AndroidBlockGuardPolicy.onReadFromDisk(StrictMode.java:1504)
        at libcore.io.BlockGuardOs.read(BlockGuardOs.java:253)
        at libcore.io.IoBridge.read(IoBridge.java:501)
        at java.io.FileInputStream.read(FileInputStream.java:307)
        at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._loadMore(UTF8StreamJsonParser.java:220)
        at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._loadMoreGuaranteed(UTF8StreamJsonParser.java:2372)
        at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._finishString2(UTF8StreamJsonParser.java:2457)
        at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._finishAndReturnString(UTF8StreamJsonParser.java:2437)
        at com.fasterxml.jackson.core.json.UTF8StreamJsonParser.getText(UTF8StreamJsonParser.java:293)
        at com.fasterxml.jackson.databind.deser.std.BaseNodeDeserializer.deserializeObject(JsonNodeDeserializer.java:267)
        at com.fasterxml.jackson.databind.deser.std.BaseNodeDeserializer.deserializeArray(JsonNodeDeserializer.java:437)
        at com.fasterxml.jackson.databind.deser.std.BaseNodeDeserializer.deserializeObject(JsonNodeDeserializer.java:261)
        at com.fasterxml.jackson.databind.deser.std.BaseNodeDeserializer.deserializeArray(JsonNodeDeserializer.java:437)
        at com.fasterxml.jackson.databind.deser.std.BaseNodeDeserializer.deserializeObject(JsonNodeDeserializer.java:261)
        at com.fasterxml.jackson.databind.deser.std.JsonNodeDeserializer.deserialize(JsonNodeDeserializer.java:68)
        at com.fasterxml.jackson.databind.deser.std.JsonNodeDeserializer.deserialize(JsonNodeDeserializer.java:15)
        at com.fasterxml.jackson.databind.ObjectMapper._readTreeAndClose(ObjectMapper.java:4254)
        at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:2692)
        at org.nypl.simplified.accounts.json.AccountProviderDescriptionCollectionParser.parse(AccountProviderDescriptionCollectionParser.kt:87)
        at org.nypl.simplified.accounts.source.nyplregistry.AccountProviderSourceNYPLRegistry.fetchDiskResults(AccountProviderSourceNYPLRegistry.kt:186)
        at org.nypl.simplified.accounts.source.nyplregistry.AccountProviderSourceNYPLRegistry.load(AccountProviderSourceNYPLRegistry.kt:98)
        at org.nypl.simplified.accounts.registry.AccountProviderRegistry.refresh(AccountProviderRegistry.kt:76)
        at org.nypl.simplified.ui.settings.SettingsFragmentAccountRegistry$onCreateView$1.onClick(SettingsFragmentAccountRegistry.kt:192)
        at android.view.View.performClick(View.java:6597)
        at android.view.View.performClickInternal(View.java:6574)
        at android.view.View.access$3100(View.java:778)
        at android.view.View$PerformClick.run(View.java:25885)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

**Comment by io7m** [comment]
Wow! That definitely shouldn't be happening. We've taken such excessive care from day one not to do any kind of I/O on the UI thread, and there we are doing it directly... :blush: 

I'm very much in favour of enabling the strictest possible strict mode everywhere.

File: simplified-main/src/main/java/org/nypl/simplified/main/MainApplication.kt [hunk 1]
```diff
@@ -69,6 +73,30 @@ class MainApplication : MultiDexApplication() {
...
 
+  /**
+   * StrictMode is a developer tool which detects things you might be doing by accident and
+   * brings them to your attention so you can fix them.
+   *
+   * StrictMode is most commonly used to catch accidental disk or network access on the
+   * application's main thread, where UI operations are received and animations take place.
+   */
+
+  private fun configureStringMode() {
```
Comment (io7m) [inline simplified-main/src/main/java/org/nypl/simplified/main/MainApplication.kt:84]: `StringMode` -> `StrictMode` :laughing:

Comment (twaddington) [reply simplified-main/src/main/java/org/nypl/simplified/main/MainApplication.kt:84]: Wow, that's embarrassing! I should not be doing this so late at night 😂

**Comment by twaddington** [comment]
> Wow! That definitely shouldn't be happening. &mdash; @io7m 

It's possible some of these _may be_ false positives. There will also be some things, like disk reads to load shared preferences, that are on the order of milliseconds and can safely be ignored.

**Comment by io7m** [comment]
That code is my fault. I think I just forgot that the `refresh` method is synchronous. I think other call sites call it on a background thread. It's definitely one that needs fixing; it's a long-running network request.

**Comment by twaddington** [comment]
> That code is my fault. I think I just forgot that the `refresh` method is synchronous. I think other call sites call it on a background thread. It's definitely one that needs fixing; it's a long-running network request. &mdash; @io7m 

Yeah, the reason I noticed is that newer versions of Android throw an exception called `NetworkOnMainThreadException` when this occurs. I'm not positive what version of Android this was first implemented on but I'm testing on API 28.

**Pull request merged**